packages/typescript-estree/src/ts-estree/ts-estree.ts

```diff
@@ -52,20 +52,80 @@ export interface BaseNode {
  * They are not included in the `Node` union below on purpose because they
  * are not ever included as part of the standard AST tree.
  */
-
-export interface Token extends BaseNode {
-  type: AST_TOKEN_TYPES;
+interface BaseToken extends BaseNode {
   value: string;
-  regex?: {
+}
+
+export interface BooleanToken extends BaseToken {
+  type: AST_TOKEN_TYPES.Boolean;
+}
+
+export interface IdentifierToken extends BaseToken {
+  type: AST_TOKEN_TYPES.Identifier;
+}
+
+export interface JSXIdentifierToken extends BaseToken {
+  type: AST_TOKEN_TYPES.JSXIdentifier;
+}
+
+export interface JSXTextToken extends BaseToken {
+  type: AST_TOKEN_TYPES.JSXText;
+}
+
+export interface KeywordToken extends BaseToken {
+  type: AST_TOKEN_TYPES.Keyword;
+}
+
+export interface NullToken extends BaseToken {
+  type: AST_TOKEN_TYPES.Null;
+}
```

Comment:
does this actually exist?
I think that we parse `null` as
```js
{
  type: 'Keyword',
  value: 'null',
}
```

Reply:
```
  if ('originalKeywordKind' in token && token.originalKeywordKind) {
    if (token.originalKeywordKind === SyntaxKind.NullKeyword) {
      return AST_TOKEN_TYPES.Null;
    } else if (
      token.originalKeywordKind >= SyntaxKind.FirstFutureReservedWord &&
      token.originalKeywordKind <= SyntaxKind.LastKeyword
    ) {
      return AST_TOKEN_TYPES.Identifier;
    }
    return AST_TOKEN_TYPES.Keyword;
  }
```